### ISSUE TITLE
Endianess flip on EventData parse for uint8[16]

### ISSUE THREAD

**Issue by TeePaps** [spohara79/estreamer#5]
It seems that the endianess of struct fields with type `uint8[16]` get their endian flipped when they are parsed.

For example, `initiator_ip` is of that type. When I manually check the buffer, I get:
`\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xac\x10\x0a\xca`
Which is an IP address in the form: `::ffff:172.16.10.202`

But the value in the parsed event data field is an integer with this hex representation:
`\xca\x0a\x10\xac\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00`

For other types, like `uint32`, this problem does not occur.

**Comment by spohara79** [comment]
Ah.. so, that's not an endianess issue.

They store IPv4 addresses by filling in bits 80-95 with 1’s, which yields an invalid IPv6
address.  If you notice on that.. the IP address is there on the end.  You should be able to use the `get_addr()` function in ```config.py```

**Comment by TeePaps** [comment]
Oh no I understand that they store IPv4 address in that form when representing them as IPv6. But if you look at the byte order, one is reversed and one is not. When I use get_addr on the EventData, I get `8204:10ac:ffff::` which further confirms this

**Comment by TeePaps** [comment]
I changed the following line of code in the unpack method, which fixes the endianness issue, but I'm still wrapping my head around all the code base so I just want to make sure this is the way _you_ would do it.

**Old code:**
```
elif fmt.startswith('BBB') or fmt.startswith('bbb'):
    val, buf = self.__unpack__(fmt, buf)
    setattr(self, field, sum([byte_ << (offset * 8) for offset, byte_ in enumerate(val)]))
````

**Modified code**
```
elif fmt.startswith('BBB') or fmt.startswith('bbb'):
    val, buf = self.__unpack__(fmt, buf)
    setattr(self, field, sum([byte_ << ((len(val) - offset - 1) * 8) for offset, byte_ in enumerate(val)]))
````

**Comment by spohara79** [comment]
if you reverse the byte array, it would be for all types; the way it fills it is based on the number..
e.g., uint8[16] is broken up as `uint8 == 'B'`, and then it fills a string by the number in the brackets.  so, `uint8[16] = 'B' * 16`; `uint8[32] = 'B' * 32`.  If you reverse the string (`val[::-1]`) it would apply to 16 and 32. 

Is this happening for the IntrusionEvent?  Did you look at uuid for the same record?

**Comment by TeePaps** [comment]
I'm not using IntrusionEvents so I can't say, but it was happening for other all the uint8[16] types for sure.

**Comment by spohara79** [comment]
What record(s) are you seeing it on?

**Comment by TeePaps** [comment]
I created a `Connection` type (for Connection Statistics data). Some of the fields are `ingress_zone` and `egress_zone`, but I wouldn't be sure about the correct order of those fields regardless. 

To test, I changed the `first_pkt_time` field from `uint32` to `uint8[4]`.
When the code uses `val` it gives me an invalid timestamp but when it uses `val[::-1]` it gives the same timestamp as uint32 would.
